tests/model_registry/rbac/test_mr_rbac.py

```diff
@@ -0,0 +1,299 @@
+            ForbiddenException: Expected before group addition, unexpected after
+        """
+        LOGGER.info(
+            "-----Test that a user can access to the Model Registry once added to a "
+            "group that has the permissions to access it-----"
+        )
+        assert model_registry_instance.name == MR_INSTANCE_NAME
+        model_registry_users_group = f"{MR_INSTANCE_NAME}-users"
+
+        # Verify initial access denied
+        with use_context(test_idp_user_session.user_context):
+            _, client_args = setup_mr_client(
+                model_registry_instance=model_registry_instance,
+                model_registry_namespace=model_registry_namespace,
+                admin_client=admin_client,
+            )
+
+            with pytest.raises(ForbiddenException) as exc_info:
+                _ = ModelRegistryClient(**client_args)
+            assert exc_info.value.status == 403
+
+        # Add user to group
+        group = Group(
+            client=admin_client,
+            name=model_registry_users_group,
+        )
+        group.update(
+            resource_dict={"metadata": {"name": model_registry_users_group}, "users": [test_idp_user_session.username]}
+        )
+
+        # Verify group membership
+        group = Group(client=admin_client, name=model_registry_users_group)
+        users = group.instance.get("users", []) or []
+        assert test_idp_user_session.username in users, (
+            f"User {test_idp_user_session.username} not in group {model_registry_users_group}. Current users: {users}"
+        )
+        LOGGER.info(f"Added user {test_idp_user_session.username} to {model_registry_users_group} group")
+        try:
+            # Wait for access to be granted
+            with use_context(test_idp_user_session.user_context):
+                _, client_args = setup_mr_client(
+                    model_registry_instance=model_registry_instance,
+                    model_registry_namespace=model_registry_namespace,
+                    admin_client=admin_client,
+                )
+
+                sampler = TimeoutSampler(
+                    wait_timeout=240,
+                    sleep=5,
+                    func=assert_positive_mr_registry,
+                    model_registry_instance=model_registry_instance,
+                    model_registry_namespace=model_registry_namespace,
+                    admin_client=admin_client,
+                )
+                for _ in sampler:
+                    break  # Break after first successful iteration
+                LOGGER.info("Successfully accessed Model Registry")
+        finally:
+            group = Group(
+                client=admin_client,
+                name=model_registry_users_group,
+            )
+            group.update(resource_dict={"metadata": {"name": model_registry_users_group}, "users": []})
+            LOGGER.info(f"Removed user {test_idp_user_session.username} from {model_registry_users_group} group")
```

Comment:
Why not create a fixture instead? That way, we don't have to handle clean up in tests.

Reply:
if I use a fixture the user would be added before the test starts, so I could not check in the test run that the user does not have access to MR but they will gain once they are added to the correct group